### PR TITLE
Initial support for volume stores in vic-machine

### PR DESCRIPTION
**Pull request by gigawhitlocks**
This change allows the creation of volume stores at install time.

### CODE REVIEW

**Comment by gigawhitlocks** [comment]
Killing the build for a minute so that I can push a couple of cleanup commits without thrashing Drone. I'll kick off a new build when I start to get reviews.

File: lib/install/management/appliance.go [hunk 1]
```diff
@@ -631,3 +632,17 @@ func (d *Dispatcher) makeSureApplianceRuns(conf *metadata.VirtualContainerHostCo
 
 	return fmt.Errorf("could not obtain IP address information from appliance: %s", updateErr)
 }
+
+func (d *Dispatcher) createVolumeStores(conf *metadata.VirtualContainerHostConfigSpec) error {
```
Comment (emlin) [inline lib/install/management/appliance.go]: the appliance.go file is for appliance VM related operations. Would suggest move to other file, store_files, where is for the datastore file deletion right now, but can be extended for more datastore files operation.

**Comment by emlin** [comment]
Everything else LGTM. 👍

**Comment by matthewavery** [comment]
lgtm

File: lib/install/management/store_files.go [hunk 2]
```diff
@@ -173,3 +173,17 @@ func (d *Dispatcher) getVCHRootDir(vchVM *vm.VirtualMachine) (string, error) {
...
 	return path.Join(parent, uuid), nil
 }
+
+func (d *Dispatcher) createVolumeStores(conf *metadata.VirtualContainerHostConfigSpec) error {
+	for _, url := range conf.VolumeLocations {
+		ds, err := d.session.Finder.Datastore(d.ctx, url.Host)
+		if err != nil {
+			return errors.Errorf("Could not retrieve datastore with host %s due to error %s", url.Host, err)
+		}
+		_, err = vsphere.NewDatastore(d.ctx, d.session, ds, url.Path)
```
Comment (fdawg4l) [inline lib/install/management/store_files.go]: We want the result of this persisted.  Specifically, the `RootURL` in this struct.

File: lib/install/validate/validator.go [hunk 3]
```diff
@@ -228,12 +228,32 @@ func (v *Validator) storage(ctx context.Context, input *data.Data, conf *metadat
...
 	// TODO: add volume locations
+	for label, volDSpath := range input.VolumeLocations {
 
+		dsURL, _, err := v.DatastoreHelper(ctx, volDSpath)
+		v.NoteIssue(err)
+		if err != nil {
+			log.Errorf("Validator could not look up volume datastore due to error: %s", err)
+			return
+		}
+		if conf.VolumeLocations == nil {
```
Comment (fdawg4l) [inline lib/install/validate/validator.go]: Can we move this check and the `make` to above the `for`?

**Comment by fdawg4l** [comment]
LGTM.  Minor comment.

File: lib/install/validate/validator.go (hunk 3 above)
Comment (gigawhitlocks) [inline lib/install/validate/validator.go]: Yes

**Comment by gigawhitlocks** [comment]
poke

**Pull request merged**